wa/utils/serializer.py

```diff
@@ -305,6 +305,9 @@ def _read_pod(fh, fmt=None):
         return json.load(fh)
     elif fmt == 'py':
         return python.load(fh)
+    elif fmt == '':
+        # Special case of no given file extension
+        raise ValueError('Please specify a file format for {} e.g. ".yaml"'.format(getattr(fh, 'name', '<none>')))
```

Comment:
I think this would make more sense to check after the splitext call, because then you can also have a more specific message, e.g. "Could not determine format from file extension; Please specify it with 'fmt' parameter"